src/Image/Image.php

```
@@ -144,9 +144,9 @@ public function html(array $attr = []): string
 	/**
 	 * Returns the PHP imagesize array
 	 */
-	public function imagesize(): array
+	public function imagesize(): array|false
 	{
-		return getimagesize($this->root);
+		return @getimagesize($this->root);
```

`Question`: @afbora Do we need the `@` to suppress errors? Or would it work without as well. The docs say `On failure, false is returned.`

---

> Errors/Exceptions[ ¶](https://www.php.net/manual/en/function.getimagesize.php#refsect1-function.getimagesize-errors)
> If accessing the filename image is impossible getimagesize() will generate an error of level [E_WARNING](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-warning). On read error, getimagesize() will generate an error of level [E_NOTICE](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-notice).

It says it will throw an error or exception if it can't read the file. If we've already done this check elsewhere, we can remove it here, as you mentioned.

However, if this line still works when there's an error in the file (which it appears to), I think using `@` would be the correct option. Or, we could put it in a try-catch block and return false.

---

I would prefer the try/catch block

---

We cannot try/catch PHP warnings and notices (that only works because Whoops converts them to exception, but will not work if Whoops is disabled). But we could use `Helpers::handleErrors()`.

However I'm not sure if it's overkill. What would we do in the handler aside from returning `false` (which the current implementation already does)?

---

That's correct, we need to catch it with `::handleErrors()`. But as @lukasbestle said, we'll return false after catching it. If we're going to do the same thing with `@`, I don't think there's any need to overcomplicate the code.